src/modules/Lan.rb

```diff
@@ -717,11 +719,17 @@ def yast_config
     # Reads system configuration
     #
     # It resets already read configuration.
+    #
+    # @return [Boolean] Returns whether the configuration was read.
     def read_config
-      system_config = Y2Network::Config.from(:wicked)
+      result = Y2Network::Config.from(:wicked)
+      system_config = result.config
       system_config.backend = NetworkService.cached_name
       Yast::Lan.add_config(:system, system_config)
       Yast::Lan.add_config(:yast, system_config.copy)
+      return true unless result.issues?
+
+      Y2Issues.report(result.issues) == :yes
```

Comment:
hmm, I am not sure if there it should not reset back system and yast config? as no basically means that partial configuration is read and replace previous one, not?

Reply:
Yes, you are right. I have adapted the method to replace the configuration only when it is accepted by the user.